# HG changeset patch
# User Punarbasu Purkayastha <ppurka@gmail.com>
# Date 1351317477 -28800
# Node ID 85a00cf7f2d8f090c4af99100d9216878f24cb33
# Parent  dd62fb8faca64c3f0c885f7627755fae5e308e3b
add RDF, CDF to linear algebra tutorial

diff --git a/doc/en/tutorial/tour_linalg.rst b/doc/en/tutorial/tour_linalg.rst
--- a/doc/en/tutorial/tour_linalg.rst
+++ b/doc/en/tutorial/tour_linalg.rst
@@ -103,6 +103,21 @@
     [ 1.00000000000000 0.000000000000000]
     [0.000000000000000  1.00000000000000]
 
+For computing eigenvalues and eigenvectors of matrices over floating
+point real or complex numbers, the matrix should be defined over ``RDF``
+(Real Double Field) or ``CDF`` (Complex Double Field), respectively. If no
+ring is specified and floating point real or complex numbers are used then
+by default the matrix is defined over the ``RR`` or ``CC`` fields,
+respectively, which do not support these computations for all the cases::
+
+    sage: ARDF = matrix(RDF, [[1.2, 2], [2, 3]])
+    sage: ARDF.eigenvalues()
+    [-0.0931712199461, 4.29317121995]
+    sage: ACDF = matrix(CDF, [[1.2, I], [2, 3]])
+    sage: ACDF.eigenvectors_right()
+    [(0.881845698329 - 0.820914065343*I, [(0.750560818381, -0.616145932705 + 0.238794153033*I)], 1),
+    (3.31815430167 + 0.820914065343*I, [(0.145594698293 + 0.37566908585*I, 0.915245825866)], 1)]
+
 Matrix spaces
 -------------
 
